### ISSUE TITLE
implement a validation loss

### ISSUE THREAD

**Issue by fanskyer** [buriburisuri/speech-to-text-wavenet#18]
thanks for sharing the implementation, tested and works well on the training set but seems overfit to it, and resulted a bad generalization on testing set. do you have any observation on that? 
I was then trying to implement to check the validation loss or validation WER, however, i was stuck on that logit.sg_reuse, because of the definition of the network is through x, residual block, ..., there are quite limited resources online regarding the sugartensor, could you point me an example how to build a validation module? thanks!

**Comment by buriburisuri** [comment]
@fanskyer Sorry for late reply.  ( enjoyed long holidays )

Exactly, my model was overfitted and need aggressive augmentation or more data if available.

Function sg_resue() is for simple network.  If you need more complex graph, use normal tensorflow way ( naming each node ).  This example will be helpful to you.  https://github.com/buriburisuri/sugartensor/tree/master/sugartensor/example

Thank you.
